### PR TITLE
Support buildArgs for image building

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2157

/cc @openshift/openshift-team-developer-productivity-test-platform @vrutkovs

### CODE REVIEW

File: pkg/steps/project_image.go [hunk 1]
```diff
@@ -31,7 +31,14 @@ func (s *projectDirectoryImageBuildStep) Inputs() (api.InputDefinition, error) {
 	return nil, nil
 }
 
-func (*projectDirectoryImageBuildStep) Validate() error { return nil }
+func (s *projectDirectoryImageBuildStep) Validate() error {
+	for i, e := range s.config.BuildArgs {
+		if e.ValueFrom != nil {
+			fmt.Errorf("build_args[%d] not implemented: valueFrom", i)
```
Comment (alvaroaleman) [inline pkg/steps/project_image.go:37]:
```suggestion
			fmt.Errorf("build_args[%d]: valueFrom must be set", i)
```

Is the implication that `Value` must be unset? If so, we should validate that. Do you have a reference for this?

Comment (hongkailiu) [reply pkg/steps/project_image.go]: We support only `value` for now.

Confirmed by @vrutkovs 

https://coreos.slack.com/archives/CBN38N3MW/p1619182185023200?thread_ts=1619083815.472400&cid=CBN38N3MW

Comment (hongkailiu) [reply pkg/steps/project_image.go]:
```
oc explain build.spec.strategy.dockerStrategy.buildArgs
KIND:     Build
VERSION:  build.openshift.io/v1

RESOURCE: buildArgs <[]Object>

DESCRIPTION:
     buildArgs contains build arguments that will be resolved in the Dockerfile.
     See https://docs.docker.com/engine/reference/builder/#/arg for more
     details.

     EnvVar represents an environment variable present in a Container.

FIELDS:
   name	<string> -required-
     Name of the environment variable. Must be a C_IDENTIFIER.

   value	<string>
     Variable references $(VAR_NAME) are expanded using the previous defined
     environment variables in the container and any service environment
     variables. If a variable cannot be resolved, the reference in the input
     string will be unchanged. The $(VAR_NAME) syntax can be escaped with a
     double $$, ie: $$(VAR_NAME). Escaped references will never be expanded,
     regardless of whether the variable exists or not. Defaults to "".

   valueFrom	<Object>
     Source for the environment variable's value. Cannot be used if value is not
     empty.
```

**Review by alvaroaleman — APPROVED** [review]

File: pkg/api/types.go [hunk 2]
```diff
@@ -1359,6 +1360,10 @@ type ProjectDirectoryImageBuildInputs struct {
 	// that will populate the build context for the Dockerfile or
 	// alter the input image for a multi-stage build.
 	Inputs map[string]ImageBuildInputs `json:"inputs,omitempty"`
+
+	// BuildArgs contains build arguments that will be resolved in the Dockerfile.
+	// See https://docs.docker.com/engine/reference/builder/#/arg for more details.
```
Comment (alvaroaleman) [inline pkg/api/types.go:1365]: Please mention here that only `Value` is supported

Comment (hongkailiu) [reply pkg/api/types.go:1365]: Updated.

File: pkg/api/types.go [hunk 3]
```diff
@@ -1359,6 +1360,11 @@ type ProjectDirectoryImageBuildInputs struct {
...
+	// Currently only value is supported.
+	BuildArgs []corev1.EnvVar `json:"build_args,omitempty"`
```
Comment (stevekuznetsov) [inline pkg/api/types.go:1367]: Use a type that exposes the options we *actually* use, not some superset that misleads users and we need to validate, please.

**Review by stevekuznetsov — COMMENTED** [review]
This needs tests. I don't think e2e, but unit for sure.

File: pkg/api/types.go (hunk 3 above)
Comment (hongkailiu) [reply pkg/api/types.go]: Updated.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**